contracts/protocol/v1/proxies/P1LiquidatorProxy.sol

```diff
@@ -0,0 +1,285 @@
+        P1Types.Balance memory currentBalance = perpetual.getAccountBalance(msg.sender);
+
+        // Get the liquidated amount and fee amount.
+        (uint256 liqAmount, uint256 feeAmount) = _getLiquidatedAndFeeAmount(
+            perpetual,
+            isBuy,
+            initialBalance,
+            currentBalance
+        );
+
+        // Transfer fee from sender to insurance fund.
+        perpetual.withdraw(msg.sender, address(this), feeAmount);
+        perpetual.deposit(_INSURANCE_FUND_, feeAmount);
+
+        // Log the result.
+        emit LogLiquidatorProxyUsed(
+            msg.sender,
+            liquidatee,
+            isBuy,
+            liqAmount,
+            feeAmount
+        );
+
+        return liqAmount;
+    }
+
+    // ============ Admin Functions ============
+
+    /**
+     * @dev Allows the owner to set the insurance fund address. Emits the LogInsuranceFundSet event.
+     *
+     * @param  insuranceFund  The address to set as the insurance fund.
+     */
+    function setInsuranceFund(
+        address insuranceFund
+    )
+        external
+        onlyOwner
+    {
+        _INSURANCE_FUND_ = insuranceFund;
+        emit LogInsuranceFundSet(insuranceFund);
+    }
+
+    // ============ Helper Functions ============
+
+    /**
+     * @dev Calculate (and verify) the maximum amount to liquidate based on the maxPosition input.
+     */
+    function _getMaxPositionDelta(
+        P1Types.Balance memory initialBalance,
+        bool isBuy,
+        SignedMath.Int memory maxPosition
+    )
+        private
+        pure
+        returns (SignedMath.Int memory)
+    {
+        SignedMath.Int memory result = maxPosition.signedSub(initialBalance.getPosition());
+
+        require(
+            result.isPositive == isBuy && result.value > 0,
+            "Cannot liquidate past maxPosition"
+        );
+
+        return result;
+    }
+
+    /**
+     * @dev Perform the liquidation by constructing the correct arguments and sending it to the
+     * perpetual.
+     */
+    function _doLiquidation(
+        I_PerpetualV1 perpetual,
+        address liquidatee,
+        SignedMath.Int memory maxPositionDelta
+    )
+        private
+    {
+        // Create accounts.
+        bool takerFirst = address(msg.sender) < liquidatee;
+        address[] memory accounts = new address[](2);
+        uint256 takerIndex = takerFirst ? 0 : 1;
+        uint256 makerIndex = takerFirst ? 1 : 0;
+        accounts[takerIndex] = msg.sender;
+        accounts[makerIndex] = liquidatee;
+
+        // Create trade args.
+        I_PerpetualV1.TradeArg[] memory trades = new I_PerpetualV1.TradeArg[](1);
+        trades[0] = I_PerpetualV1.TradeArg({
+            takerIndex: takerIndex,
+            makerIndex: makerIndex,
+            trader: _LIQUIDATION_,
+            data: abi.encode(
+                maxPositionDelta.value,
+                maxPositionDelta.isPositive,
+                false // allOrNothing,
```

Comment:
small nit: remove `,`

Reply:
removed the code, but will fix in the original place the code was copied from